prototype/core/impl/src/main/java/org/eclipse/sensinact/prototype/impl/SessionManager.java

```diff
@@ -50,6 +56,16 @@ public class SessionManager implements SensiNactSessionManager, TypedEventHandle
 
     private final Map<String, String> userDefaultSessionIds = new HashMap<>();
 
+    @Activate
+    void activate() {
+        metrics.registerGauge("sensinact.sessions", () -> sessions.size());
+    }
+
+    @Deactivate
+    void deactivate() {
+        metrics.unregisterGauge("sensinact.sessions");
```

Comment:
The listener pattern (which this is an example of) can be a challenge in OSGi. Could this be turned into a whiteboard?